utils/parts.stanza

```diff
@@ -389,6 +391,32 @@ defn string-or-unknown (json: JSON) -> String|UNKNOWN :
     (s: JNull) :
       UNKNOWN
 
+public defenum Category :
+  capacitor
+  connector
+  diode
+  fuse
+  inductor
+  microcontroller
+  oscillator
+  power_supply_chip
+  resistor
+  transistor
+
+defn category-or-unknown (category-str: String) -> Category|UNKNOWN :
+  switch(category-str):
+    "capacitor": capacitor
+    "connector": connector
+    "diode": diode
+    "fuse": fuse
+    "inductor": inductor
+    "microcontroller": microcontroller
+    "oscillator": oscillator
+    "power_supply_chip": power_supply_chip
+    "resistor": resistor
+    "transistor": transistor
+    else: UNKNOWN
```

Comment:
Hey, here is a compact way to write it, it is less readable but has lower maintenance / less error prone:
```
import lang-utils

val category-values = map(Category, 0 to Category-length)
val string-to-category-value = to-hashtable(to-string, category-values)

defn category-or-unknown (category-str: String) -> Category|UNKNOWN :
  get?(string-to-category-value, category-str, UNKNOWN)
```

For OCDB, preferring readability like you did makes sense, so not a change request

Reply:
Oh awesome thanks! I was wondering if there was a better way...

I'll leave as is for the reasons you stated.